controlplane/kubeadm/internal/controllers/controller.go

```diff
@@ -138,7 +138,7 @@ func (r *KubeadmControlPlaneReconciler) SetupWithManager(ctx context.Context, mg
 
 	r.controller = c
 	r.recorder = mgr.GetEventRecorderFor("kubeadmcontrolplane-controller")
-	r.ssaCache = ssa.NewCache()
+	r.ssaCache = ssa.NewCache("controlplane/kubeadm")
```

Comment:
```suggestion
	r.ssaCache = ssa.NewCache("kubeadmcontrolplane")
```

Sorry should have mentioned that before.

The controller name from a CR perspective is either:
* the parameter of the `Named` method (if used)
* or the lower case kind of what is passed into `For` (`KubeadmControlPlane` => `kubeadmcontrolplane`)

Controller-runtime uses that for log k/v pairs and for metrics. So we are usually aligning to that for consistency wherever possible

Reply:
ah! cool, thanks for clarifying -- pushed up this fix 👍